tabs.js

```diff
@@ -0,0 +1,126 @@
+var h = require('hyperscript')
+var u = require('./lib/util')
+  each = u.each,
+  find = u.find,
+  isVisible = u.isVisible,
+  setVisible = u.setVisible,
+  setInvisible = u.setInvisible
+
+function toggle_focus(page) {
+  isVisible(page)
+    ? blur(page)
+    : focus(page)
+}
+
+function focus(page) {
+  if (isVisible(page)) return
+
+  setVisible(page)
+  var el = page.firstChild
+  el.dispatchEvent(new CustomEvent('focus', {target: el}))
+}
+
+function blur (page) {
+  if (!isVisible(page)) return
+
+  setInvisible(page)
+  var el = page.firstChild
+  el.dispatchEvent(new CustomEvent('blur', {target: el}))
+}
+
+function moveTo(page, content, i) {
+  if(!content.children.length || i >= content.children.length)
+    content.appendChild(page)
+  else
+    content.insertBefore(page, content.children[i])
+}
+
+module.exports = function (content, onSelect) {
+  var tabs = h('section', { className: '.tabs' })
+  var selection
+
+  function build_tab (page) {
+    var link = h('a', {
+      className: '.link',
```

Comment:
oh, maybe what i need most is a clarification on how classes are used. before I was following BEM http://getbem.com/introduction/ (@dust's suggestion)

it stands for Block Element Modifier. it looks like mcss is similar, but Capital case means block, `-hypen` means modifier... and `.class` means element.

hmm...
except in mcss the element is always coupled to a tag? 
you can have `Item { div.main { ... }}` but not `Item { .main {...} }`
is that really an advantage? @mmckegg ?